### PR TITLE
Red Chapter 18 Answers

### PR DESCRIPTION
**Pull request by utopia-2**

### CODE REVIEW

File: red/Animal/lib/RaceHorse.pm [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+dbmopen( our %HORSES, 'racehorses', 0666 )
+  or croak 'Failed to open racehorse database with read/write permissions';
```
Comment (alexdeas) [inline red/Animal/lib/RaceHorse.pm:21]: This seems neat, I'll demand an explanation in the meeting probably.

File: red/Animal/lib/RaceHorse.pm [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    use RaceHorse;
+
+    my $horse = RaceHorse->new( name => 'Existential Crisis Pony' );
+    $horse->won;
+
+=head1 SUBROUTINES/METHODS
+
+=cut
+
+sub valid_properties { qw( wins places shows losses ) }
```
Comment (alexdeas) [inline red/Animal/lib/RaceHorse.pm:36]: Avoiding constants?

File: red/Animal/lib/RaceHorse.pm [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+=head2 won/placed/showed/losses
+
+Used to record the outcome of a single race.
+
+=cut
+
+sub won { $incr->( shift, 'wins' ) }
+sub placed { $incr->( shift, 'places' ) }
+sub showed { $incr->( shift, 'shows' ) }
+sub losses { $incr->( shift, 'losses' ) }
```
Comment (alexdeas) [inline red/Animal/lib/RaceHorse.pm:73]: So your fancy blessed closure made this part harder then.

File: red/Animal/lib/RaceHorse.pm (hunk 2 above)
Comment (utopia-2) [inline red/Animal/lib/RaceHorse.pm:36]: Nah, need this due to me showing that you can more heavily protect instance state in the previous version. I was tempted to rip it out and start again, but figured it'd be more interesting to try and get everything working.

**Pull request merged**